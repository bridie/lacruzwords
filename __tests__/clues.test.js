import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme'
import { currentTheme, acrossWords, downWords } from './fixtures/currentTheme';
import Clues from './../src/components/clues';

function setup() {
	const props = { currentTheme }

	const enzymeWrapper = shallow(<Clues {...props} />)

	return {
		props,
		enzymeWrapper
	}
}

describe("Clues Component", () => {
	it('renders correctly', () => {
		const { props } = setup();

  		const clues = renderer.create(<Clues {...props} />).toJSON();
  		expect(clues).toMatchSnapshot();
	});

	it('correctly filters the across words', () => {
		const { enzymeWrapper } = setup();
		expect(enzymeWrapper.instance().getWordsWithDirection('across')).toEqual(acrossWords);
	});

	it('correctly filters the down words', () => {
		const { enzymeWrapper } = setup();
		expect(enzymeWrapper.instance().getWordsWithDirection('down')).toEqual(downWords);
	});
})

