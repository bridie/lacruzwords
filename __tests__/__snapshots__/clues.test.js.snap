// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clues Component renders correctly 1`] = `
<div
  className="clues"
>
  <div
    className="clues__column clues__column--across"
  >
    <h2>
      Across
    </h2>
    <p>
      <span>
        1
        .
      </span>
      <span>
         
      </span>
      dog
    </p>
    <p>
      <span>
        3
        .
      </span>
      <span>
         
      </span>
      crocodile
    </p>
    <p>
      <span>
        6
        .
      </span>
      <span>
         
      </span>
      spider
    </p>
    <p>
      <span>
        7
        .
      </span>
      <span>
         
      </span>
      cat
    </p>
    <p>
      <span>
        9
        .
      </span>
      <span>
         
      </span>
      elephant
    </p>
  </div>
  <div
    className="clues__column clues__column--down"
  >
    <h2>
      Down
    </h2>
    <p>
      <span>
        2
        .
      </span>
      <span>
         
      </span>
      rhinoceros
    </p>
    <p>
      <span>
        4
        .
      </span>
      <span>
         
      </span>
      turtle
    </p>
    <p>
      <span>
        5
        .
      </span>
      <span>
         
      </span>
      monkey
    </p>
    <p>
      <span>
        8
        .
      </span>
      <span>
         
      </span>
      rabbit
    </p>
  </div>
</div>
`;
